### PR TITLE
Implement onboarding

### PR DESCRIPTION
**Pull request by MazeXP**
Implements https://github.com/discord/discord-api-docs/pull/5915

### CODE REVIEW

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Guilds/IOnboardingPrompt.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    /// <summary>
+    /// Gets a value indicating whether the prompt is required before a user completes the onboarding flow.
+    /// </summary>
+    bool IsRequired { get; }
+
+    /// <summary>
+    /// Gets a value indicating whether the prompt is present in the onboarding flow.
+    /// </summary>
+    /// <remarks>If false, the prompt will only appear in the Channels & Roles tab.</remarks>
```
Comment (Nihlus) [inline Backend/Remora.Discord.API.Abstractions/API/Objects/Guilds/IOnboardingPrompt.cs:68]: Literal ampersands are invalid in XML comments - use the XML-escaped variant `&amp;`.

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Guilds/IPromptOption.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+namespace Remora.Discord.API.Abstractions.Objects;
+
+/// <summary>
+/// Represents an option for a prompt.
+/// </summary>
+[PublicAPI]
+public interface IPromptOption
+{
+    /// <summary>
+    /// Gets the ID of the prompt.
```
Comment (Nihlus) [inline Backend/Remora.Discord.API.Abstractions/API/Objects/Guilds/IPromptOption.cs:36]: Should say "ID of the prompt option".

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Guilds/IPromptOption.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    /// </summary>
+    Snowflake ID { get; }
+
+    /// <summary>
+    /// Gets a list of IDs for channels assigned to a member when the option is selected.
```
Comment (Nihlus) [inline Backend/Remora.Discord.API.Abstractions/API/Objects/Guilds/IPromptOption.cs:41]: Should say "channels a member is added to".

File: Backend/Remora.Discord.API/Extensions/ServiceCollectionExtensions.cs [hunk 4]
```diff
@@ -668,6 +668,18 @@ private static JsonSerializerOptions AddGuildObjectConverters(this JsonSerialize
         options.AddDataObjectConverter<IWelcomeScreen, WelcomeScreen>();
         options.AddDataObjectConverter<IWelcomeScreenChannel, WelcomeScreenChannel>();
         options.AddDataObjectConverter<IGuildThreadQueryResponse, GuildThreadQueryResponse>();
+        options.AddDataObjectConverter<IGuildOnboarding, GuildOnboarding>()
+            .WithPropertyName(o => o.DefaultChannelIDs, "default_channel_ids")
```
Comment (Nihlus) [inline Backend/Remora.Discord.API/Extensions/ServiceCollectionExtensions.cs:672]: This override should not be required.

File: Backend/Remora.Discord.API/Extensions/ServiceCollectionExtensions.cs [hunk 5]
```diff
@@ -668,6 +668,18 @@ private static JsonSerializerOptions AddGuildObjectConverters(this JsonSerialize
...
+            .WithPropertyName(o => o.IsEnabled, "enabled");
+
+        options.AddDataObjectConverter<IOnboardingPrompt, OnboardingPrompt>()
+            .WithPropertyName(p => p.IsSingleSelect, "single_select")
+            .WithPropertyName(p => p.IsRequired, "required")
+            .WithPropertyName(p => p.IsInOnboarding, "in_onboarding");
+
+        options.AddDataObjectConverter<IPromptOption, PromptOption>()
+            .WithPropertyName(o => o.ChannelIDs, "channel_ids")
```
Comment (Nihlus) [inline Backend/Remora.Discord.API/Extensions/ServiceCollectionExtensions.cs:681]: This override should not be required.

File: Backend/Remora.Discord.API/Extensions/ServiceCollectionExtensions.cs [hunk 6]
```diff
@@ -668,6 +668,18 @@ private static JsonSerializerOptions AddGuildObjectConverters(this JsonSerialize
...
+            .WithPropertyName(o => o.RoleIDs, "role_ids");
```
Comment (Nihlus) [inline Backend/Remora.Discord.API/Extensions/ServiceCollectionExtensions.cs:682]: This override should not be required.

File: Backend/Remora.Discord.Caching/API/CachingDiscordRestGuildAPI.cs [hunk 7]
```diff
@@ -1093,6 +1093,33 @@ public Task<Result<IChannel>> CreateGuildStageChannelAsync
...
     }
 
+    /// <inheritdoc />
+    public async Task<Result<IGuildOnboarding>> GetGuildOnboardingAsync
+    (
+        Snowflake guildID,
+        CancellationToken ct = default
+    )
+    {
+        var key = new KeyHelpers.GuildWelcomeScreenCacheKey(guildID);
```
Comment (Nihlus) [inline Backend/Remora.Discord.Caching/API/CachingDiscordRestGuildAPI.cs:1103]: Incorrect cache key type - use the `GuildOnboardingCacheKey` method you added.

File: Backend/Remora.Discord.API/Extensions/ServiceCollectionExtensions.cs (hunk 4 above)
Comment (MazeXP) [reply Backend/Remora.Discord.API/Extensions/ServiceCollectionExtensions.cs:672]: I tested without it and tests started to fail. Therefore it is required. I suppose its cause of the "...IDs" part.

Comment (Nihlus) [reply Backend/Remora.Discord.API/Extensions/ServiceCollectionExtensions.cs:672]: Interesting - will look into this later.

**Review by Nihlus — APPROVED** [review]

File: Backend/Remora.Discord.API/Extensions/ServiceCollectionExtensions.cs (hunk 4 above)
Comment (MazeXP) [reply Backend/Remora.Discord.API/Extensions/ServiceCollectionExtensions.cs:672]: Must be the SnakeCaseNamingPolicy that is making things up.

**Pull request merged**